share/functions/__fish_complete_zfs_rw_properties.fish

```diff
@@ -0,0 +1,100 @@
+function __fish_complete_zfs_rw_properties -d "Completes with ZFS read-write properties"
+	set -l OS ""
+	if passwd --help >/dev/null ^&1
+		set OS "Linux"
+	else # Not Linux, so use the ugly uname
+		set -l os_type (uname)
+		switch $os_type
+			case Darwin
+				set OS "macOS"
+			case FreeBSD
+				set OS "FreeBSD"
+			case SunOS
+				set OS "SunOS"
+			# Others?
+			case "*"
+				set OS "unknown"
+		end
+	end
+	echo -e "aclinherit\t"(_ "Inheritance of ACL entries")" (discard, noallow, restricted, passthrough, passthrough-x)"
+	echo -e "atime\t"(_ "Update access time on read")" (on, off)"
+	echo -e "canmount\t"(_ "Is the dataset mountable")" (on, off, noauto)"
+	set -l additional_algs ''
+    if test $OS = 'FreeBSD'; or test $OS = 'SunOS'
```

Comment:
Oh, your syntax is much better; I was just unsure how to make mine prettier.

Reply:
Should I also use `contains` for other alike tests, or can I let `test`?

Reply:
You can, sure. It's a matter of taste. Here I find it nicer because it makes working with more than one possibility easier. For just one OS, I think I personally would lean towards `test`, unless of course the list can change, in which case `contains` is easier to adapt.

Pick whichever you prefer.

Reply:
Let's just say it's fine as it is.